test/e2e_node/garbage_collector_test.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	dockertypes "github.com/docker/engine-api/types"
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	docker "k8s.io/kubernetes/pkg/kubelet/dockertools"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+/*
+GarbageCollect tests that the Kubelet conforms to the Kubelet Garbage Collection Policy, found here:
+http://kubernetes.io/docs/admin/garbage-collection/
+TODO (dashpole): Once the Container Runtime Interface (CRI) is complete, generalize to other runtimes (other than docker)
+TODO (dashpole): Once dynamic config is possible, test different values for maxPerPodContainer and maxContainers
+*/
+const (
+	defaultDockerEndpoint                = "unix:///var/run/docker.sock"
+	defaultRuntimeRequestTimeoutDuration = 2 * time.Minute
+	maxPerPodContainer                   = 2
+	maxTotalContainers                   = -1
+
+	runtimePollInterval = 10 * time.Second
+	deleteTimeout       = 2 * time.Minute
+
+	containerNamePrefix = "gc-test-container"
```

Comment:
It would be better to randomize the container name prefix, so that each test won't affect each other.

Currently the container left in previous `It` will affect following test.